src/ngx_http_lua_headers_out.c

```diff
@@ -488,14 +488,17 @@ ngx_http_lua_set_output_header(ngx_http_request_t *r, ngx_http_lua_ctx_t *ctx,
     ngx_http_lua_header_val_t         hv;
     ngx_http_lua_set_header_t        *handlers = ngx_http_lua_set_handlers;
     ngx_uint_t                        i;
+    ngx_int_t                         rc;
 
     dd("set header value: %.*s", (int) value.len, value.data);
 
-    if (ngx_http_lua_check_unsafe_string(r, key.data, key.len,
-                                         "header name") != NGX_OK
-        || ngx_http_lua_check_unsafe_string(r, value.data, value.len,
-                                            "header value") != NGX_OK)
-    {
+    rc = ngx_http_lua_copy_escape_header(r, &key, 1);
+    if (rc != NGX_OK) {
```

Comment:
We don't need to introduce the `rc` variable here since it's always used only once. So better be:

```C
if (ngx_http_lua_copy_escape_header(r, &key, 1) != NGX_OK) {
    return NGX_ERROR;
}
```

The same applies to the `if` statement below.